In [1]:
import geopandas as gpd
import pandas as pd


# Prepare clean 2017 file

In [2]:
# Import the starting "clean" version of the 2017 file
org_2017 = ".\\poljine_SHP\\ZV2017_d96tm_clean.shp"
# Import the resulting HAND dataset for the year 2017
hand_2017 = ".\\poljine_SHP\\ZV2017_d96tm_pc075_hand.csv"

In [17]:
# Read files
df_clean17 = gpd.read_file(org_2017)
df_hand17 = pd.read_csv(hand_2017)

In [18]:
df_clean17.tail()

index  GERK_PID  RABA_ID  POVR_GERK_  POLJINA_ID SIFRA_KMRS OBRAZEC  \
680693  873390   1531986     1300  15676.4145     3133665        204      GP   
680694  873392   5234722     1222   1021.1447     3133667        699      GP   
680695  873399   5944379     1211   5816.1295     4079093        100      GP   
680696  873400   4836379     1100   6689.5872     4079099        020      GP   
680697  873401   4836379     1100   6689.5872     4079101        809      GP   

           POVR_ar  INTERSECT  \
680693  156.767387          0   
680694   10.211659          0   
680695   58.162621          0   
680696   16.568304          0   
680697   23.181401          0   

                                                 geometry  
680693  POLYGON ((531591.349 129010.053, 531591.562 12...  
680694  POLYGON ((531593.914 129091.364, 531583.766 12...  
680695  POLYGON ((419381.441 74175.221, 419383.523 741...  
680696  POLYGON ((418528.359 79292.935, 418652.938 793...  
680697  POLYGON ((418429.496 79231.686, 418506.709 792...

In [19]:
df_hand17.tail()

Unnamed: 0  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_  SIFRA_KMRS  \
664133      681156   1892284     2976726     1300   8760.9728         204   
664134      681157   3588000     2976727     1300   7807.7304         204   
664135      681158   1646627     2976728     1300   7559.2166         204   
664136      681159   3588612     2976729     1300   6120.1665         204   
664137      681160   3588639     2976730     1300   5015.4655         204   

       OBRAZEC  INTERSECT    POVR_ar  REGIJA  pix_count  hand_mean   hand_std  
664133      GP          0  87.609728       2         92  18.987577  10.833754  
664134      GP          0  78.077304       2         81  23.950169   8.193017  
664135      GP          0  75.592166       2         80  77.466759  24.283596  
664136      GP          0  61.201665       2         65   9.352770   5.397253  
664137      GP          0  50.154655       2         51   9.910322   2.914534

The `ZV2017_d96tm_clean.shp` file has extra lines that have not been included in the final table. These correspond to polygons hat are narrow and were dropped during the "buffernig" step of prreparing the shapefile.

In [27]:
# This will filter the "clean" file to include only relevant lines

shp_2017 = ".\\poljine_SHP\\ZV2017_d96tm_pc075_hand.shp"
df_2017 = gpd.read_file(shp_2017)
df_2017.tail()

GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_ SIFRA_KMRS OBRAZEC  \
664133   1892284     2976726     1300   8760.9728        204      GP   
664134   3588000     2976727     1300   7807.7304        204      GP   
664135   1646627     2976728     1300   7559.2166        204      GP   
664136   3588612     2976729     1300   6120.1665        204      GP   
664137   3588639     2976730     1300   5015.4655        204      GP   

        INTERSECT    POVR_ar  REGIJA  pix_count  hand_mean   hand_std  \
664133          0  87.609728       2         92  18.987577  10.833754   
664134          0  78.077304       2         81  23.950169   8.193017   
664135          0  75.592166       2         80  77.466759  24.283596   
664136          0  61.201665       2         65   9.352770   5.397253   
664137          0  50.154655       2         51   9.910322   2.914534   

                                                 geometry  
664133  POLYGON ((434921.542 108572.694, 434922.432 10...  
664134  POLYGON ((434747.327 108244.216, 434730.310 10...  
664135  POLYGON ((432921.938 108920.374, 432931.875 10...  
664136  POLYGON ((435669.945 108166.760, 435723.973 10...  
664137  POLYGON ((435690.513 108261.462, 435639.992 10...

Drop columns from the source file (this shape was generated from a HAND dataset, that was created earlier in the project - I have never saved the "clean" version of the file, therfore I had to recreate it from one of the tables)

In [28]:
df_2017 = df_2017.drop(columns=["pix_count", "hand_mean", "hand_std"])
df_2017.tail()

GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_ SIFRA_KMRS OBRAZEC  \
664133   1892284     2976726     1300   8760.9728        204      GP   
664134   3588000     2976727     1300   7807.7304        204      GP   
664135   1646627     2976728     1300   7559.2166        204      GP   
664136   3588612     2976729     1300   6120.1665        204      GP   
664137   3588639     2976730     1300   5015.4655        204      GP   

        INTERSECT    POVR_ar  REGIJA  \
664133          0  87.609728       2   
664134          0  78.077304       2   
664135          0  75.592166       2   
664136          0  61.201665       2   
664137          0  50.154655       2   

                                                 geometry  
664133  POLYGON ((434921.542 108572.694, 434922.432 10...  
664134  POLYGON ((434747.327 108244.216, 434730.310 10...  
664135  POLYGON ((432921.938 108920.374, 432931.875 10...  
664136  POLYGON ((435669.945 108166.760, 435723.973 10...  
664137  POLYGON ((435690.513 108261.462, 435639.992 10...

In [29]:
# Add unique index
df_2017 = df_2017.reset_index()
df_2017 = df_2017.rename(columns={"index": "id_17"})
df_2017

id_17  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_ SIFRA_KMRS OBRAZEC  \
0            0   2537988     2104906     1100  19308.9997        801      GP   
1            1   2537988     2104908     1100  19308.9997        005      GP   
2            2   2537988     2104910     1100  19308.9997        203      GP   
3            3   1143420     2105003     1100   8277.3796        801      GP   
4            4   3077991     2652377     1211  14987.7976        100      GP   
...        ...       ...         ...      ...         ...        ...     ...   
664133  664133   1892284     2976726     1300   8760.9728        204      GP   
664134  664134   3588000     2976727     1300   7807.7304        204      GP   
664135  664135   1646627     2976728     1300   7559.2166        204      GP   
664136  664136   3588612     2976729     1300   6120.1665        204      GP   
664137  664137   3588639     2976730     1300   5015.4655        204      GP   

        INTERSECT     POVR_ar  REGIJA  \
0               0   57.930456       5   
1               0   30.520097       5   
2               0   26.799442       5   
3               0   40.952670       5   
4               0  149.877976       4   
...           ...         ...     ...   
664133          0   87.609728       2   
664134          0   78.077304       2   
664135          0   75.592166       2   
664136          0   61.201665       2   
664137          0   50.154655       2   

                                                 geometry  
0       POLYGON ((547508.347 86829.736, 547545.426 868...  
1       POLYGON ((547497.850 86785.076, 547578.847 867...  
2       POLYGON ((547599.139 86850.730, 547670.666 868...  
3       POLYGON ((547165.958 85941.243, 547179.112 859...  
4       POLYGON ((419052.089 80240.803, 419056.851 802...  
...                                                   ...  
664133  POLYGON ((434921.542 108572.694, 434922.432 10...  
664134  POLYGON ((434747.327 108244.216, 434730.310 10...  
664135  POLYGON ((432921.938 108920.374, 432931.875 10...  
664136  POLYGON ((435669.945 108166.760, 435723.973 10...  
664137  POLYGON ((435690.513 108261.462, 435639.992 10...  

[664138 rows x 11 columns]

In [31]:
# Save clean
df_2017.to_file(".\\poljine_SHP\\ZV2017_d96tm_clean_ids.shp")

This is the final version of a clean ZV2017 shapefile. It includes:
- 664137 lines
- unique index called `id_17`

# Prepare clean 2018 and 2019 files

In [7]:
# 2018
df_18 = gpd.read_file(".\\poljine_SHP\\ZV2018_d96tm_clean.shp")
df_18

index  GERK_PID  RABA_ID  POVR_GERK_  POLJINA_ID SIFRA_KMRS OBRAZEC  \
0            0   2439387     1300  31637.2652     3280114        204      GP   
1            1   4896286     1100   1187.8658     3280115        405      GP   
2            2   4896302     1300  31987.1334     3280116        204      GP   
3            3   2439736     1300   7075.6430     3280117        204      GP   
4            4   2439393     1300   5647.2772     3280118        204      GP   
...        ...       ...      ...         ...         ...        ...     ...   
680693  873390   1531986     1300  15676.4145     3133665        204      GP   
680694  873392   5234722     1222   1021.1447     3133667        699      GP   
680695  873399   5944379     1211   5816.1295     4079093        100      GP   
680696  873400   4836379     1100   6689.5872     4079099        020      GP   
680697  873401   4836379     1100   6689.5872     4079101        809      GP   

           POVR_ar  INTERSECT  \
0       316.379280          0   
1        11.878907          0   
2       319.878034          0   
3        70.757911          0   
4        56.473954          0   
...            ...        ...   
680693  156.767387          0   
680694   10.211659          0   
680695   58.162621          0   
680696   16.568304          0   
680697   23.181401          0   

                                                 geometry  
0       POLYGON ((506260.639 158941.631, 506270.058 15...  
1       POLYGON ((506784.582 159034.079, 506784.811 15...  
2       POLYGON ((506750.306 159062.243, 506749.725 15...  
3       POLYGON ((506818.507 158884.252, 506819.626 15...  
4       POLYGON ((506907.476 159065.017, 506909.868 15...  
...                                                   ...  
680693  POLYGON ((531591.349 129010.053, 531591.562 12...  
680694  POLYGON ((531593.914 129091.364, 531583.766 12...  
680695  POLYGON ((419381.441 74175.221, 419383.523 741...  
680696  POLYGON ((418528.359 79292.935, 418652.938 793...  
680697  POLYGON ((418429.496 79231.686, 418506.709 792...  

[680698 rows x 10 columns]

In [13]:
# Add unique index
new_df_18 = df_18.drop(columns=["index"])
new_df_18 = new_df_18.reset_index()
new_df_18 = new_df_18.rename(columns={"index": "id_18"})
new_df_18

id_18  GERK_PID  RABA_ID  POVR_GERK_  POLJINA_ID SIFRA_KMRS OBRAZEC  \
0            0   2439387     1300  31637.2652     3280114        204      GP   
1            1   4896286     1100   1187.8658     3280115        405      GP   
2            2   4896302     1300  31987.1334     3280116        204      GP   
3            3   2439736     1300   7075.6430     3280117        204      GP   
4            4   2439393     1300   5647.2772     3280118        204      GP   
...        ...       ...      ...         ...         ...        ...     ...   
680693  680693   1531986     1300  15676.4145     3133665        204      GP   
680694  680694   5234722     1222   1021.1447     3133667        699      GP   
680695  680695   5944379     1211   5816.1295     4079093        100      GP   
680696  680696   4836379     1100   6689.5872     4079099        020      GP   
680697  680697   4836379     1100   6689.5872     4079101        809      GP   

           POVR_ar  INTERSECT  \
0       316.379280          0   
1        11.878907          0   
2       319.878034          0   
3        70.757911          0   
4        56.473954          0   
...            ...        ...   
680693  156.767387          0   
680694   10.211659          0   
680695   58.162621          0   
680696   16.568304          0   
680697   23.181401          0   

                                                 geometry  
0       POLYGON ((506260.639 158941.631, 506270.058 15...  
1       POLYGON ((506784.582 159034.079, 506784.811 15...  
2       POLYGON ((506750.306 159062.243, 506749.725 15...  
3       POLYGON ((506818.507 158884.252, 506819.626 15...  
4       POLYGON ((506907.476 159065.017, 506909.868 15...  
...                                                   ...  
680693  POLYGON ((531591.349 129010.053, 531591.562 12...  
680694  POLYGON ((531593.914 129091.364, 531583.766 12...  
680695  POLYGON ((419381.441 74175.221, 419383.523 741...  
680696  POLYGON ((418528.359 79292.935, 418652.938 793...  
680697  POLYGON ((418429.496 79231.686, 418506.709 792...  

[680698 rows x 10 columns]

In [14]:
# Save CSV clean
new_df_18.to_file(".\\poljine_SHP\\ZV2018_d96tm_clean_ids.shp")

## 2019

In [10]:
# 2019
df_19 = gpd.read_file(".\\poljine_SHP\\KMRS2019_d96tm_clean.shp")
df_19

index  GERK_PID  POLJINA_ID SIFRA_KMRS  \
0            0   2228346     4924022        204   
1            5   3479338     4089010        204   
2            7   5667433     4089012        204   
3            8   5990311     4089013        204   
4           10   5990298     4924025        204   
...        ...       ...         ...        ...   
626057  820140   5134669     5074724        204   
626058  820142   5328914     5074726        204   
626059  820143   3728216     5074727        206   
626060  820149   5986203     5074733        201   
626061  820150   1651059     5074941        204   

                                                 geometry  
0       POLYGON ((431524.725 89609.706, 431535.835 896...  
1       POLYGON ((432007.736 89259.174, 432007.757 892...  
2       POLYGON ((432714.241 88480.527, 432714.409 884...  
3       POLYGON ((431793.459 89455.808, 431784.999 894...  
4       POLYGON ((431903.276 89477.617, 431895.815 894...  
...                                                   ...  
626057  POLYGON ((409122.729 41039.861, 409126.508 410...  
626058  POLYGON ((408130.129 41425.027, 408127.444 414...  
626059  POLYGON ((408410.623 42371.535, 408415.263 423...  
626060  POLYGON ((407848.549 41825.277, 407893.186 418...  
626061  POLYGON ((410741.888 109210.137, 410742.961 10...  

[626062 rows x 5 columns]

In [11]:
# Add unique index
new_df_19 = df_19.drop(columns=["index"])
new_df_19 = new_df_19.reset_index()
new_df_19 = new_df_19.rename(columns={"index": "id_19"})
new_df_19

id_19  GERK_PID  POLJINA_ID SIFRA_KMRS  \
0            0   2228346     4924022        204   
1            1   3479338     4089010        204   
2            2   5667433     4089012        204   
3            3   5990311     4089013        204   
4            4   5990298     4924025        204   
...        ...       ...         ...        ...   
626057  626057   5134669     5074724        204   
626058  626058   5328914     5074726        204   
626059  626059   3728216     5074727        206   
626060  626060   5986203     5074733        201   
626061  626061   1651059     5074941        204   

                                                 geometry  
0       POLYGON ((431524.725 89609.706, 431535.835 896...  
1       POLYGON ((432007.736 89259.174, 432007.757 892...  
2       POLYGON ((432714.241 88480.527, 432714.409 884...  
3       POLYGON ((431793.459 89455.808, 431784.999 894...  
4       POLYGON ((431903.276 89477.617, 431895.815 894...  
...                                                   ...  
626057  POLYGON ((409122.729 41039.861, 409126.508 410...  
626058  POLYGON ((408130.129 41425.027, 408127.444 414...  
626059  POLYGON ((408410.623 42371.535, 408415.263 423...  
626060  POLYGON ((407848.549 41825.277, 407893.186 418...  
626061  POLYGON ((410741.888 109210.137, 410742.961 10...  

[626062 rows x 5 columns]

In [12]:
# Save CSV clean
new_df_19.to_file(".\\poljine_SHP\\KMRS2019_d96tm_clean_ids.shp")